packages/context/src/Callback/Callback.container.js

```diff
@@ -1,63 +1,41 @@
-import { withRouter } from "react-router-dom";
-import { compose, withProps, lifecycle, withHandlers, pure } from "recompose";
+import React, { useEffect, useCallback } from 'react';
+import { withRouter } from 'react-router-dom';
 
-import { getUserManager, oidcLog } from "../Services";
-import CallbackComponent from "./Callback.component";
+import { getUserManager, oidcLog, withServices } from '../Services';
+import CallbackComponent from './Callback.component';
 
-let _isCalled = false;
-const _setCalling = val => {
-  _isCalled = val;
-};
-
-export const onRedirectSuccess = ({ history }) => user => {
-  oidcLog.info("Successfull Callback");
+export const onRedirectSuccess = (history, oidcLogInternal) => user => {
+  oidcLogInternal.info('Successfull login Callback');
   if (user.state.url) {
```

Comment:
what happen if user.state not exist ?

Reply:
We can't be at this point if we don't have state. OidcClient throw an error if the request back without state